Added 1-Click Deploy tutorial (#160)
DefangLabs/defang-docs: docs/tutorials/using-one-click-deploy.md

---
title: Using 1-Click Deploy
sidebar_position: 800
description: Use 1-Click Deploy to easily deploy a sample to the Defang Playground. 
---

# Using 1-Click Deploy

This tutorial will show you how to use Defang 1-Click Deploy to deploy a sample. 

> The 1-Click Deploy button is intended to be a quickstart way for new users to deploy to the [Defang Playground](/docs/concepts/defang-playground) with no prior installation needed. 

Reviewer: no prior installation of what - CLI? can we clarify?

Reviewer: Also, it should be "for new users to deploy a sample to the..." - correct?

:::info

> If this is not your first time, we recommend using the [Defang CLI](/docs/getting-started) to get a full experience of deploying with Defang. 

Reviewer: Replace with something like: To access the full range of features provided by Defang, we recommend using the [Defang CLI](/docs/getting-started).

:::

## Step 1 - Choose a Sample
Head to our [list of samples](https://defang.io/#samples) and click a sample you want to deploy. Then, click on the button that says "1-Click Deploy".

Alterntively, you can find the "1-Click Deploy" button located in the `README.md` file of each sample's repository. 

## Step 2 - Allow Permissions

After you've clicked, you will be prompted to use GitHub to log in. Once you see a "Create a Repository" page appear, allow the sample repository to get cloned into your GitHub account. 

## Step 3 - Wait for Deployment to Complete

> A Github action workflow will automatically start running to install Defang and deploy the sample to the Defang Playground. If you wish to, you can see this by going into the "Actions" tab in your GitHub repository. 

Reviewer: I believe actions should be capitalized in GitHub Actions - please correct all instances.

You can view the status of your deployment in the [Defang Portal](https://portal.defang.dev/), or by downloading the [Defang CLI](/docs/getting-started).  

:::tip
Defang has a tutorial on how you can [monitor your services](/docs/tutorials/monitoring-your-services). 
:::

:::info
If you decide to make a commit later to a repository created from 1-Click Deploy, then the project will automatically get deployed again to Defang Playground.
:::

### Configuration in 1-Click Deploy

> If the sample you chose requires setting configuration, such as API keys, you can set sensitive config values as secrets in your GitHub repository and the action will automatically deploy them for you. 

Reviewer: The last part is a little unclear " and the action will automatically deploy them for you."
The action is the GitHub action - correct? We should be specific.
What is the "them" - also be specific.

---
title: Using 1-Click Deploy
sidebar_position: 300
description: Use 1-Click Deploy to easily deploy a sample to the Defang Playground. 
---

# Using 1-Click Deploy

This tutorial will show you how to use Defang 1-Click Deploy to deploy a sample to the Defang Playground. 

> The 1-Click Deploy button is intended to be a quickstart way for new users to deploy to the [Defang Playground](/docs/concepts/defang-playground) with no prior installation of the CLI needed. 

Reviewer 2: What do you think about this? "quickstart way" felt a little ambiguous.
```suggestion
The 1-Click Deploy button is intended to be a quickstart guide to deploying an application to the [Defang Playground](/docs/concepts/defang-playground) with no prior installation of the CLI needed.
```

Author: The button itself isn't a guide (as in an explicit tutorial) though, it's almost like a shortcut. Also, I think we should keep the word "new users" somewhere to make the distinction extremely clear between the other tutorial.

Author: I do agree we can use a better wording than "way"

Reviewer 2: Ohh duh. Once again, not reading in the mindset of the reader.Maybe it would be better to come on a bit more confidently, and say something like this, but it's not super important. 

```suggestion
The 1-Click Deploy button is the easiest way for new users to deploy your project to the [Defang Playground](/docs/concepts/defang-playground). No CLI installation is required. 
```

Author: This sentence is part of the tutorial for brand-new users, so it should not say "for new users to deploy your project". I agree with the other changes here though.

:::info
To access the full range of features provided by Defang, we recommend using the [Defang CLI](/docs/getting-started).
:::

## Step 1 - Choose a Sample
Head to our [list of samples](https://defang.io/#samples) and click a sample you want to deploy. Then, click on the button that says "1-Click Deploy".

![one-click-deploy-button](/img/use-one-click-tutorial/one-click-deploy-button.png)

:::info
Alternatively, you can find the "1-Click Deploy" button located in the `README.md` file of each sample's GitHub repository. 

![deploy-with-defang-button](/img/use-one-click-tutorial/deploy-with-defang-button.png)
:::

## Step 2 - Allow Permissions

After you've clicked, you will be prompted to use GitHub to log in. Once you see a "Create a new repository" page appear, allow the sample repository to get cloned into your GitHub account. You can do this by clicking the green "Create repository" button at the bottom. 

![create-repository](/img/use-one-click-tutorial/create-repository.png)

## Step 3 - Wait for Deployment to Complete

> A Github Action workflow will automatically start running to install Defang and deploy the sample to the Defang Playground. If you wish to, you can see this by going into the "Actions" tab in your GitHub repository. 

Reviewer 2: Nitpick, but I think it might be better to be more terse and avoid filler like "if you wish"

```suggestion
A Github Action workflow will automatically start running to install Defang and deploy the sample to the Defang Playground. You can see this by going into the "Actions" tab in your GitHub repository. 
```

Author: maybe i can lead it with "For interest's sake, you can see this..." ? I do want to make it clear that it's not an action they have to take in order for them to complete the tutorial

Reviewer 2: Personally, I think it's unnecessary to hedge it like that. They should probably go see it for themselves, but I also think "You can see..." is enough.

Either way, if you feel strongly about leaving it as-is, that's OK. No need to hold this up over this detail.
